activesupport/lib/active_support/string_inquirer.rb

```diff
@@ -9,6 +9,34 @@ module ActiveSupport
   #
   #   Rails.env.production?
   class StringInquirer < String
+
+    def initialize(value, valid_values = [])
```

Comment:
The point of using `define_method` is that it's much faster. Isn't it a concern? In the other hand, I wouldn't expect long lists to check against...